W-16244513 Adding new info about enabling Einstein for ACB (#341)
mulesoft/docs-code-builder: modules/ROOT/pages/troubleshoot-generative-ai.adoc

= Troubleshoot Einstein for Anypoint Code Builder

Errors sometimes occur when you use the Einstein for Anypoint Code Builder.

== Unexpected Query Results When Using the Einstein Generative Flow Feature

> If the generative flow is not returning accurate and useful results for the prompts you enter, rewrite your prompt and follow the guidelines xref:int-create-integrations-ai.adoc#craft-ai-prompt[for crafting an effective prompt].

Reviewer: ```suggestion
If the generative flow is not returning accurate and useful results for the prompts you enter, rewrite your prompt and follow the guidelines to xref:int-create-integrations-ai.adoc#craft-ai-prompt[craft an effective prompt].
```

> == Access Errors

Reviewer: Any way to make this heading more specific and unique?

Troubleshoot Einstein Access Errors?

> If you can't access Einstein, contact your Anypoint Platform organization administrator. If you are the organization administrator, these are the things to check:

Reviewer: ```suggestion
If you can't access Einstein, contact your Anypoint Platform organization administrator. If you are the organization administrator, check these things:
```

> * Did you accept the Terms and Conditions and xref:int-ai-enable-einstein.adoc[enable Einstein for Anypoint Code Builder] for your organization? 

Reviewer 2: should we capitalize "Terms and Conditions" as we do here or leave lower case, as above in another file? Looks like it's "terms and conditions" most everywhere in the ACB docs, except for some image alt text

> * The Salesforce organization you are connected to for Einstein requests xref:https://help.salesforce.com/s/articleView?id=sf.generative_ai_enable.htm&type=5[have Einstein turned on]? 

Reviewer: ```suggestion
* The Salesforce organization you are connected to for Einstein requests xref:https://help.salesforce.com/s/articleView?id=sf.generative_ai_enable.htm&type=5[has Einstein enabled]. 
```

Reviewer: All statements or all questions...

* The Salesforce organization you are connected to for Einstein requests has an xref:xref:access-management::trusted-salesforce-org.adoc[established tenant relationship] with your Anypoint Platform organization, and the connection is enabled.  

> * Is the Salesforce organization you are connected to for Einstein requests a valid organization? For example, verify that it's not expired or otherwise disabled. 

Reviewer: ```suggestion
* The Salesforce organization you are connected to for Einstein requests is valid. For example, verify that it's not expired or otherwise disabled. 
```

> * Is the Salesforce organization you are connected to for Einstein request a valid organization? For example, verify that it's not expired or otherwise disabled. 

Reviewer 2: ```suggestion
* Is the Salesforce organization you are connected to for Einstein requests a valid organization? For example, verify that it's not expired or otherwise disabled. 
```

> == Authorization Errors

Reviewer: More context?

Reviewer: Troubleshoot Anypoint Platform Authorization Errors?

> If you receive an authorization error, contact your Anypoint Platform organization administrator. If you are the organization administrator, these are the things to check:

Reviewer: ```suggestion
If you receive an authorization error, contact your Anypoint Platform organization administrator. If you are the organization administrator, check these things:
```

Reviewer: or ", check the following:"?

> * The user has the {PERMISSION} assigned in Anypoint Platform access management? 

Reviewer: ```suggestion
* The user has the {PERMISSION} assigned in Anypoint Platform access management.
```

Reviewer: Font for {PERMISSION}? (oh, this is a variable...?)

> * Does the user have the {PERMISSION} assigned in Anypoint Platform access management? 

Reviewer 2: assuming {PERMISSION}  is or will be defined somewhere or spelled out here later. Is it a local attribute?

> * Did you accept the Einstein for Anypoint Code Builder Terms and Conditions?

Reviewer 2: same with T&C: It that the title of the terms and conditions agreement?

Reviewer 2: Do we need to point them to the page explaining how they can accept T&C?

> * Your Anypoint Platform organization is current and active. 

Reviewer: How do you check this?

> == Einstein Does Not Appear for EU Cloud Users on the Desktop IDE

Reviewer: ```suggestion
== Einstein Doesn't Appear for EU Cloud Users on the Desktop IDE
```

* xref:int-create-integrations-ai.adoc#before-you-begin[You have the required permissions].
* You completed the xref:start-acb.adoc#change-clouds[Set the Desktop IDE to the EU or US Cloud] procedure.

> == The Einstein Generative Flow Feature Isn't Working

Reviewer: Can this be:
== Einstein Generative Flow Isn't Working

> The Einstein Generative Flow feature works only when you have one project at a time open. It doesn't work if you have multiple projects open concurrently. 

Reviewer 2: Wondering if this will change in October. Something to keep on the radar screen.

Author: Yes, definitely.
